github/repos_hooks_test.go

```diff
@@ -296,3 +296,261 @@ func TestRepositoriesService_TestHook_invalidOwner(t *testing.T) {
+		"commits": [
+			{
+			"added":   ["1", "2", "3"],
+			"author":{
+				"email": "abc@gmail.com",
+				"name": "abc",
+				"username": "abc_12"
+			},
+			"committer": {
+				"email": "abc@gmail.com",
+				"name": "abc",
+				"username": "abc_12"
+			}, 
+			"id":       "1",
+			"message":  "WebHookCommit",
+			"modified": ["abc", "efg", "erd"],
+			"removed":  ["cmd", "rti", "duv"]
+			}
+		],
+		"compare": "compare",
+		"created": true,
+		"forced":  false,
+		"head_commit": {
+			"added":   ["1", "2", "3"],
+		"author":{
+			"email": "abc@gmail.com",
+			"name": "abc",
+			"username": "abc_12"
+		},
+		"committer": {
+			"email": "abc@gmail.com",
+			"name": "abc",
+			"username": "abc_12"
+		}, 
+		"id":       "1",
+		"message":  "WebHookCommit",
+		"modified": ["abc", "efg", "erd"],
+		"removed":  ["cmd", "rti", "duv"]
+		},
+		"installation": {
+			"id": 12
+		},
+		"organization": {
+			"id" : 22
+		},
+		"pusher":{
+			"login": "rd@yahoo.com",
+			"id": 112
+		},
+		"repository":{
+			"id": 321,
+			"node_id": "node_321"
+		},
+		"sender":{
+			"login": "st@gmail.com",
+			"id": 202
+		}
+	}`
+
+	testJSONMarshal(t, v, want)
+}
+
+func TestBranchWebHookAuthor_Marshal(t *testing.T) {
+	testJSONMarshal(t, &WebHookAuthor{}, "{}")
+
+	v := &WebHookAuthor{
+		Email:    String("abc@gmail.com"),
+		Name:     String("abc"),
+		Username: String("abc_12"),
+	}
+
+	want := `{
+			"email": "abc@gmail.com",
+			"name": "abc",
+			"username": "abc_12"
+	}`
+
+	testJSONMarshal(t, v, want)
+}
+
+func TestBranchWebHookCommit_Marshal(t *testing.T) {
+	testJSONMarshal(t, &WebHookPayload{}, "{}")
+
+	v := &WebHookCommit{
+		Added: []string{"1", "2", "3"},
+		Author: &WebHookAuthor{
+			Email:    String("abc@gmail.com"),
+			Name:     String("abc"),
+			Username: String("abc_12"),
+		},
+		Committer: &WebHookAuthor{
+			Email:    String("abc@gmail.com"),
+			Name:     String("abc"),
+			Username: String("abc_12"),
+		},
+		ID:       String("1"),
+		Message:  String("WebHookCommit"),
+		Modified: []string{"abc", "efg", "erd"},
+		Removed:  []string{"cmd", "rti", "duv"},
+	}
+
+	want := `{
+		"added":   ["1", "2", "3"],
+		"author":{
+			"email": "abc@gmail.com",
+			"name": "abc",
+			"username": "abc_12"
+		},
+		"committer": {
+			"email": "abc@gmail.com",
+			"name": "abc",
+			"username": "abc_12"
+		}, 
+		"id":       "1",
+		"message":  "WebHookCommit",
+		"modified": ["abc", "efg", "erd"],
+		"removed":  ["cmd", "rti", "duv"]
+	}`
+
+	testJSONMarshal(t, v, want)
+}
+
+func TestBranchcreateHookRequest_Marshal(t *testing.T) {
+	testJSONMarshal(t, &WebHookAuthor{}, "{}")
+
+	v := &createHookRequest{
+		Name:   "abc",
+		Events: []string{"1", "2", "3"},
+		Active: Bool(true),
+		Config: map[string]interface{}{
+			"thing": "@123",
+		},
+	}
+
+	want := `{
+		"name": "abc",
+		"active": true,
+		"events": ["1","2","3"],
+		"config":{
+			"thing": "@123"
+		}
+	}`
+
+	testJSONMarshal(t, v, want)
+}
+func TestBranchHook_Marshal(t *testing.T) {
```

Comment:
Please add blank line between tests (before this line).

Reply:
added thanks